sdk-base/src/main/java/com/mapbox/maps/plugin/viewport/ViewportPlugin.kt

```diff
@@ -13,69 +13,99 @@ import com.mapbox.maps.plugin.viewport.transition.DefaultViewportTransition
-   * and to the final [ViewportState] when ended.
+   * When called, the [status] goes to [ViewportTransition] and to the final [ViewportState] when ended.
+   *
+   * Transitioning to state 'x' when the current [status] is 'ViewportStatus.State(x)' invokes [completionListener]
+   * synchronously with true and does not modify [ViewportPlugin.status].
    *
-   * If transition is canceled, state goes to IDLE.
+   * Transitioning to state 'x' when the current [status] is 'ViewportStatus.Transition(_, x)' invokes
+   * [completionListener] synchronously with false and does not modify [ViewportPlugin.status].
+   *
+   * If transition is failed or canceled, [status] goes to [ViewportStatus.Idle].
    *
    * @param targetState The target [ViewportState] to transition to.
-   * @param transition The [ViewportTransition] that's used to transition to target state, if not specified, the [ViewportPlugin.defaultTransition] will be used.
-   * @param completionListener The listener to observe the completion state.
+   * @param transition The [ViewportTransition] that's used to transition to target state, if not specified, the [ViewportPlugin.defaultTransition] will be used. Defaults to null.
+   * @param completionListener The listener that's invoked when the transition ends, defaults to null.
    */
   fun transitionTo(targetState: ViewportState, transition: ViewportTransition? = null, completionListener: CompletionListener? = null)
 
   /**
-   * Immediately goes to IDLE state canceling all ongoing transitions.
+   * Sets [ViewportPlugin.status] to [ViewportStatus.Idle] synchronously.
+   *
+   * This cancels any active [ViewportState] or [ViewportTransition].
    */
   fun idle()
 
   /**
-   * Options that impact the [ViewportPlugin].
+   * Configuration options that impact the [ViewportPlugin].
    */
   var options: ViewportOptions
 
   // Transitions
 
   /**
-   * DefaultViewportTransition with default options.
+   * [ViewportPlugin.transitionTo] uses this transition unless some non-null value is passed to its
+   * transition argument.
    *
-   * This transition is used unless overridden by one of the registered transitions.
+   * Defaults to [DefaultViewportTransition] with default options.
    */
   var defaultTransition: ViewportTransition
 
   // Observers
 
   /**
-   * Adds [ViewportStatusObserver] to observe the status change.
+   * Subscribes [ViewportStatusObserver] to observe the [ViewportStatus] change.
+   *
+   * [ViewportPlugin] keeps a strong reference to registered observers. Adding the same observer again
+   * while it is already subscribed has no effect.
+   *
+   * Note: Observers are notified of status changes asynchronously on the main queue. This means that
+   * by the time the notification is delivered, the status may have already changed again. This behaviour
+   * is necessary to allow observers to trigger further transitions while avoiding out-of-order delivery
+   * of status changed notifications.
```

Comment:
`asynchronously on the main queue` - a bit confused with `asynchronously` word here

I guess main point here is that if status did change - we will notify an observer by scheduling this update on main thread while before it is _actually_ executed - status could already be switched again. am I right here?

Reply:
yes, you are correct